Classes/TextUtils.cs

```diff
@@ -24,19 +24,106 @@ namespace RadioDld
-            // Will only remove dates with the same month & year as the programme itself, but any day of the month
-            // as there is sometimes a mismatch of a day or two between the date in a title and the publish date.
-            Regex matchStripDate = new Regex("\\A(" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/\\d{2} ?-? )?(?<name>.*?)( ?:? (\\d{2}/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "|((Mon|Tue|Wed|Thu|Fri) )?(\\d{1,2}(st|nd|rd|th)? )?(" + stripDate.ToString("MMMM", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("MMM", CultureInfo.InvariantCulture) + ")( \\d{1,2}(st|nd|rd|th)?| (" + stripDate.ToString("yy", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "))?))?\\Z");
+            // Will only remove dates within a set range eitherside of the programme date.
 
-            if (matchStripDate.IsMatch(name))
+            // Issue #59
+            // - Look for date in middle of name (as well as start and end)
+            // - look for other date formats like 13.02.2009; also 13 Feb 2009; 13th feb 2009; 13th-Feb-2009; 13feb09;
+            // - consider full day names as well eg Monday, Tuesday...
+            // - also other delims before and or after the date e.g. 'title - 2009-02-13'
+            try
             {
-                name = matchStripDate.Match(name).Groups["name"].ToString();
-            }
+                // Checks for match on ANY date in the string...
+                if (matchStripDate.IsMatch(name))
+                {
+                    // Date Found - now get the date value found
+                    string sDateFound = matchStripDate.Match(name).ToString();
+
+                    // Strip trouble characters eg ' and check for 'sept' and remove 't' also remove (st | nd | rd | th) and keep numeric
+                    sDateFound = sDateFound.Replace("'", string.Empty);
+                    sDateFound = sDateFound.Replace("Sept", "Sep").Replace("sept", "sep");
+                    sDateFound = removeDaySuffix.Replace(sDateFound, " ");
 
-            return name;
+                    // Convert to DateTime for comparison to similardate
+                    DateTime dtDateFound = Convert.ToDateTime(sDateFound, CultureInfo.CurrentCulture);
+
+                    // Check if date is within the +/- 'similardateday' range of 'date'
+                    if (dtDateFound >= stripDate.AddDays(-iSimilarDateDayRange) && dtDateFound <= stripDate.AddDays(iSimilarDateDayRange))
+                    {
+                       name = matchStripDate.Replace(name, string.Empty).ToString().Trim();
+                    }
+                }
+
+                return name;
+            }
+            catch
+            {
+                return null;
+            }
```

Comment:
There shouldn't be a blanket `try {} catch {}` around the code - this prevents errors from being reported correctly and would cause mysterious blanks instead of episode names to appear in the list.

Reply:
Haven't finished this part of the code; agree that mysterious blank are bad :) but feel i need to do something rather than just throw error.

Reply:
Actually, allowing an exception to propagate up and crash the application is absolutely the correct thing to do with errors that aren't expected, as this is how they get reported back to me via the unhandled exception handler.